src/scaffoldmaker/utils/tubemesh.py

```diff
@@ -281,71 +280,87 @@ def getCoordinatesFromInner(xInner, d1Inner, d2Inner, d3Inner,
         relativeThicknessList.append(relativeThicknessList[-1])
 
     for n2 in range(elementsCountAlong + 1):
-        wallThickness = wallThicknessList[n2]
+        if outward:
+            wallThickness = wallThicknessList[n2]
+        else:
+            wallThickness = -wallThicknessList[n2]
         for n1 in range(elementsCountAround):
             n = n2*elementsCountAround + n1
-            norm = d3Inner[n]
-            # Calculate outer coordinates
-            x = [xInner[n][i] + norm[i]*wallThickness for i in range(3)]
-            xOuter.append(x)
+            norm = d3Surf[n]
+            # Calculate extruded coordinates
+            x = [xSurf[n][i] + norm[i]*wallThickness for i in range(3)]
```

Comment:
Use the signed wall displacement here only:
```
            x = [xSurf[n][i] + norm[i]*wallOutwardDisplacement for i in range(3)]
```

Reply:
Done.